### ISSUE TITLE
Add tests to code generation

### ISSUE THREAD

**Issue by gpuente** [powerhouse-inc/powerhouse#2]
## Description

Add some tests against the code generated would help us to prevent bad generated code.
- we can include some tests against different document models and check the file content to match what is expected

**Comment by acaldas** [comment]
Hygen is run as a cli utility so it might require some work to create tests as it can't be used programmatically. It spawns a new process that writes to files. https://github.com/jondot/hygen/issues/118

We could do something similar to [Snapshot Testing](https://jestjs.io/docs/snapshot-testing), where we store the expected output and the zip file and the test checks if the generated files are the same.

**Comment by gpuente** [comment]
yes, actually that's what I was thinking. we can store some zip samples, and for running the tests we can generate the code for those zip files, and store the expected snapshots to compare.
